datafusion/physical-plan/src/stream.rs

```diff
@@ -362,6 +362,8 @@ pin_project! {
 
         #[pin]
         stream: S,
+
+        transform_schema: bool,
```

Comment:
this seems like this is fixing the symptom rather than the root cause 

I think it would be better to have the correct schema reflected in the plan in the first place 🤔